lib/chef/knife/vsphere_vm_snapshot.rb

```diff
@@ -147,17 +147,23 @@ def find_node(tree, name)
   end
 
   def display_node(node, current, shift = 1)
-    descr = node.name + ' ' + node.createTime.iso8601
-    out = ''
-    out << '+--' * shift
+    snapshot_tree = {}
+    children = []
+    unless node.childSnapshotList.empty?
+      node.childSnapshotList.each { |item| children << display_node(item, current, shift + 1) }
+    end
     if node.snapshot == current
-      out << ui.color(descr, :cyan) << '\n'
+      snapshot_tree = { 'SnapshotName' => node.name,
+                       'SnapshotDescription' => node.description,
+                       'SnapshotCreationDate' => node.createTime.iso8601,
+                       'IsCurrentSnapshot' => true,
+                       'Children' => children }
     else
```

Comment:
Since the bulk of the hash is the same you could DRY this up a bit by something like

```
snapshot_tree = { common stuff}
snapshot_tree.merge!({ 'IsCurrentSnapshot' => true}) if node.snapshot == current
```

Reply:
yup, I can  make this change :)